service/src/main/java/greencity/service/ubs/UBSManagementServiceImpl.java

```diff
@@ -905,89 +905,104 @@ private void setValueForOrderStatusIsCancelledOrDoneAsTrue(OrderStatusTranslatio
+            Integer capacity = bagRepository.findCapacityById(entry.getKey());
+            StringBuilder bagTranslation = bagTranslationRepository.findNameByBagId(entry.getKey(), languageId);
 
             if (order.getOrderStatus() == OrderStatus.ADJUSTMENT
                 || order.getOrderStatus() == OrderStatus.CONFIRMED
                 || order.getOrderStatus() == OrderStatus.FORMED
                 || order.getOrderStatus() == OrderStatus.NOT_TAKEN_OUT) {
                 Long confirmWasteWas =
-                    updateOrderRepository.getConfirmWaste(dto.get(i).getOrderId(), Long.valueOf(dto.get(i).getBagId()));
-                if (nonNull(dto.get(i).getConfirmedQuantity())
-                    && !confirmWasteWas.equals(Long.valueOf(dto.get(i).getConfirmedQuantity()))) {
-                    if (i == 0) {
+                    updateOrderRepository.getConfirmWaste(orderId, entry.getKey().longValue());
+                if (nonNull(confirmWasteWas)
+                    && !confirmWasteWas.equals(entry.getValue().longValue())) {
+                    if (countOfChanges == 0) {
                         values.append(OrderHistory.CHANGE_ORDER_DETAILS + " ");
                     }
-                    values.append(bagTranslation).append(" ").append(capacity).append(" л: ").append(confirmWasteWas)
-                        .append(" шт на ").append(dto.get(i).getConfirmedQuantity()).append(" шт.");
+                    values.append(bagTranslation).append(" ").append(capacity).append(" л: ")
+                        .append(confirmWasteWas)
+                        .append(" шт на ").append(entry.getValue()).append(" шт.");
                 }
-            } else if (order.getOrderStatus() == OrderStatus.ON_THE_ROUTE
+            }
+        }
+    }
+
+    private void collectEventAboutExportedWaste(Map<Integer, Integer> exported, Long languageId, Order order,
+        Long orderId, int countOfChanges, StringBuilder values) {
+        for (Map.Entry<Integer, Integer> entry : exported.entrySet()) {
+            Integer capacity = bagRepository.findCapacityById(entry.getKey());
+            StringBuilder bagTranslation = bagTranslationRepository.findNameByBagId(entry.getKey(), languageId);
+            if (order.getOrderStatus() == OrderStatus.ON_THE_ROUTE
                 || order.getOrderStatus() == OrderStatus.BROUGHT_IT_HIMSELF
                 || order.getOrderStatus() == OrderStatus.DONE
                 || order.getOrderStatus() == OrderStatus.CANCELED) {
-                Long exporterWasteWas = updateOrderRepository.getExporterWaste(dto.get(i).getOrderId(),
-                    Long.valueOf(dto.get(i).getBagId()));
-                if (nonNull(dto.get(i).getExportedQuantity())
-                    && !exporterWasteWas.equals(Long.valueOf(dto.get(i).getExportedQuantity()))) {
-                    if (i == 0) {
+                Long exporterWasteWas = updateOrderRepository.getExporterWaste(orderId,
```

Comment:
this condition is not covered by tests

Reply:
![image](https://user-images.githubusercontent.com/5016120/147576717-0bf786e4-8942-4d68-a072-9bd7316413ad.png)

Reply:
Got it. I will add this on the weekend after I implements all main logic